packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -495,7 +499,13 @@ function commitUnmount(current: Fiber): void {
     case ClassComponent: {
       safelyDetachRef(current);
       const instance = current.stateNode;
-      if (typeof instance.componentWillUnmount === 'function') {
+      if (
+        // Typically, a component that mounted will have an instance. However,
+        // outside of concurrent mode, a suspended component may commit without
+        // an instance, so we need to check whether it exists.
+        instance !== null &&
```

Comment:
So why do we still need this check? I thought the separate tag is supposed to address it.